File: apps/scanner/app/src/test/java/com/airferry/app/scan/QrTrackingPolicyTest.kt
package com.airferry.app.scan

import org.junit.Assert.assertEquals
import org.junit.Test

class QrTrackingPolicyTest {
    @Test
    fun completeRoiHitCancelsPartialFullScanAging() {
        var streak = 1 // periodic full scan saw only 3/4
        streak = QrTrackingPolicy.shrinkStreakAfterTrackedResult(streak, 4, 4)
        assertEquals(0, streak)

        streak += 1
        streak = QrTrackingPolicy.shrinkStreakAfterTrackedResult(streak, 3, 4)
        assertEquals(1, streak)
    }
}

File: apps/web/test/qr-tracking-policy.test.mjs
import assert from "node:assert/strict"
import test from "node:test"

import { shrinkStreakAfterTrackedResult } from "../src/lib/qr-tracking-policy.ts"

test("complete ROI hits cancel partial full-scan aging", () => {
  let streak = 0
  streak += 1 // periodic full scan saw only 3/4
  streak = shrinkStreakAfterTrackedResult(streak, 4, 4)
  assert.equal(streak, 0)

  streak += 1
  streak = shrinkStreakAfterTrackedResult(streak, 3, 4)
  assert.equal(streak, 1)
})
